src/get-default-blueprint-name-override.js

```diff
@@ -1,22 +1,32 @@
 'use strict';
 
+const fs = require('fs');
+const path = require('path');
 const { spawn } = require('./run');
 
 /**
  * Run npm view to get the package.json and parse it to check if it contains property
  * to override the default blueprint name
  *
- * @param {string} packageName - Name of package
+ * @param {string} packageNameOrPath - Name of package i.e. ember-cli-package-name or path to a local package
+ * @param {string} cwd - Default to the process's CWD. Parent functions can override
  * @returns {Promise<Array<string>>}
  */
-module.exports = async function getBlueprintNameOverride(packageName) {
-  let ps = await module.exports.spawn('npm', ['view', packageName, '--json']);
+module.exports = async function getBlueprintNameOverride(packageNameOrPath, cwd = process.cwd()) {
+  let localPackageJsonPath = path.join(path.resolve(cwd, packageNameOrPath), 'package.json');
+  let packageJsonStr;
 
-  if (ps.exitCode > 0) {
-    return null;
+  if (fs.existsSync(localPackageJsonPath)) {
+    packageJsonStr = fs.readFileSync(localPackageJsonPath);
+  } else {
+    let ps = await module.exports.spawn('npm', ['view', packageNameOrPath, '--json']);
```

Comment:
We already have a helper to deal with npm. From getVersions:

```
'use strict';

const npm = require('boilerplate-update/src/npm');

/**
 * Returns an array of strings i.e.
 * [
 *   "0.0.0",
 *   "3.23.0-beta.2"
 * ]
 * @param packageName
 * @returns {Promise<Array<string>>}
 */
module.exports = async function getVersions(packageName) {
  return await npm.json('view', packageName, 'versions');
};
```

Reply:
Switching to use helper